utils/line-directive

```diff
@@ -244,8 +245,29 @@ def run():
         dashes = sys.argv.index('--')
         sources = sys.argv[1:dashes]
 
+        if sources[0][0] == '@':
+            # Read sources from a file.
+            with open(sources[0][1:], 'r') as files:
+                sources = filter(None, files.read().split(' '))
+
+        # The first argument of command_args is the process to open.
+        # subprocess.Popen doesn't normalise arguments. This means that trying
+        # to open a non-normalised file (e.g. C:/swift/./bin/swiftc.exe) on
+        # Windows results in file/directory not found errors, as Popen delegates
+        # to the Win32 CreateProcess API. Unix systems handle non-normalised
+        # paths, so don't have this problem.
+        # Arguments passed to the process are normalised by the process.
+        command_args = sys.argv[dashes + 1:]
+        command_args[0] = os.path.normpath(command_args[0])
+
+        last_index = len(command_args) - 1
+        if command_args[last_index][0] == '@':
+            # Read sources from a file.
+            command_args = command_args[0:last_index]
+            command_args.extend(sources)
```

Comment:
Is there a reason that we want the response file to be the last entry always?  I think that we should expand any and in place.